core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -93,10 +95,12 @@ object AclCommand {
     }
   }
 
-  private def removeAcl(opts: AclCommandOptions) {
+  private[admin] def removeAcl(opts: AclCommandOptions) {
     withAuthorizer(opts) { authorizer =>
       val resourceToAcl = getResourceToAcls(opts)
 
+      validatePrincipalTypes(resourceToAcl, authorizer.getSupportedPrincipalTypes())
```

Comment:
I wonder if we should do the validation as part of `getResourceToAcls` instead of adding it in multiple places.